src/main/java/StringManipulation.java

```diff
@@ -1,32 +1,121 @@
-public class StringManipulation implements StringManipulationInterface {
+import java.util.ArrayList;
+import java.util.*;
 
+public class StringManipulation implements StringManipulationInterface {
+    
+    public String stored;
+    
     @Override
     public String getString() {
-        return null;
+        return stored;
     }
 
     @Override
     public void setString(String string) {
+        stored = string;
     }
 
     @Override
     public int count() {
-        return 0;
+        if(stored == null || stored.isEmpty()) {
+            return 0;
+        }
+        String[] words = stored.split("\\s+");
+        return words.length; 
     }
 
     @Override
     public String removeNthCharacter(int n, boolean maintainSpacing) {
-        return null;
+        //throw cases
+        if(n > stored.length()) {
+            throw new IndexOutOfBoundsException("IndexOB");
+        } else if(n <= 0) {
+            throw new IllegalArgumentException("IllegalArg");
+        }        
+        
+        String result;
+        
```

Comment:
Here is how you can do that:
// Create a mutable string (StringBuilder in Java) from the input string
    StringBuilder = create a mutable string from stored
// Loop through the multiples of n
    for i from 1 until n * i <= stored.length(); i++:
        // Calculate the index of the character to be removed/replaced. Since indexing is 0 bases, subtract 1.
        index = (n * i) - 1
        // If maintainSpacing is true, replace the character at the position with a space
        if maintainSpacing is true:
            set character at position index in StringBuilder to ' '
        // If maintainSpacing is false, remove the character at the position
        else:
            // Decrease index by i-1 to account for previously deleted characters
            delete the character at position (index - (i - 1)) in StringBuilder
    // Convert the mutable string back to a regular string and return it
    return convert StringBuilder to string